pandas/tests/indexes/test_numeric.py

```diff
@@ -919,6 +919,10 @@ def test_constructor(self):
         res = Index([1, 2**63])
         tm.assert_index_equal(res, idx)
 
+        idx = Index([-1, 2**63], dtype=object)
```

Comment:
```
In [1]: Series([np.nan],dtype='int64')
ValueError: cannot convert float NaN to integer

In [2]: Index([np.nan],dtype='int64')
Out[2]: Float64Index([nan], dtype='float64')
```

yeah this seems different that what we do in Series. I agree I think this should raise, IOW, IF you specify a dtype and its not compatible it should raise, rather than fall back. (of course if you don't specify a dtype it will just figure it out).

Note that we should raise if we have a compat type though, e.g. ``dtype=int32`` -> ``int64`` is fine (same with ``float32`` -> ``float64``).

I think we may be a bit relaxed about conversions when int is specified though. See what breaks if you change this to be more strict.